sdk/worker/task_runner.go

```diff
@@ -288,7 +296,10 @@ func (c *TaskRunner) executeAndUpdateTask(taskName string, task model.Task, exec
 	defer c.runningWorkerDone(taskName)
 	defer concurrency.HandlePanicError("execute_and_update_task " + string(task.TaskId) + ": " + string(task.Status))
 	taskResult := c.executeTask(&task, executeFunction)
-	c.updateTaskWithRetry(taskName, taskResult)
+	err := c.updateTaskWithRetry(taskName, taskResult)
+	if err != nil {
```

Comment:
Should we call `pauseOnGenericError` here?

Reply:
no, retry already does that.